plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -309,10 +315,10 @@ && predicateMatches(parquetPredicate, block, dataSource, descriptorsByPath, parq
                 internalFields.add(Optional.empty());
             }
             else {
-                internalFields.add(Optional.ofNullable(getParquetType(column, fileSchema, useColumnNames))
+                internalFields.add(Optional.ofNullable(getParquetType(column, fileSchema, columnMapping))
                         .flatMap(field -> {
-                            String columnName = useColumnNames ? column.getBaseColumnName() : fileSchema.getFields().get(column.getBaseHiveColumnIndex()).getName();
-                            return constructField(column.getBaseType(), lookupColumnByName(messageColumn, columnName));
+                            ColumnIO columnIo = getColumnIo(column, messageColumn, fileSchema, columnMapping);
```

Comment:
preexisting: I find `field` variable name rather misleading because the method actually returns a type.

The variable `field` is actually fully ignored. 
Does it make sense to refactor the code to make it more understandable?